### DISCUSSION TITLE
Pause A2DP and play a sound from SD card

### DISCUSSION THREAD

**Discussion by VincenziMichael** [pschatzmann/arduino-audio-tools#559] [Q&A] [answered]
Hi, i would like pause the bluetooth speaker by pressing a button, then play a sound effect from the sd card and then resume playback.
Can sameone help me?

I've started from this example: [player-sd_a2dp-audiokit.ino](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-audiokit/player-sd_a2dp-audiokit/player-sd_a2dp-audiokit.ino)
Changin from audiokit to a simple i2s (i'm using a MAX98357 board).

I think I am very close, starting bluetooth or sd separately works, but when I change the source get error: **register I2S object to platform failed**

Probably i can't share i2s with player and a2dp_sink? Do you know a work around to archieve this?

_To simplify the code, I set the change every 5 seconds_
```
#include "AudioTools.h"
#include "AudioLibs/AudioA2DP.h"
#include "AudioLibs/AudioSourceSDFAT.h"
#include "AudioCodecs/CodecMP3Helix.h"

const char *startFilePath="/";
const char* ext="mp3";
I2SStream i2s;
AudioSourceSDFAT source(startFilePath, ext);
MP3DecoderHelix decoder;
AudioPlayer player(source, i2s, decoder);
BluetoothA2DPSink a2dp_sink;
bool player_active = false;
bool bt_active = true;

// switch between a2dp and sd player
void changeMode() {
  player_active = !player_active;
  if (player_active){
      Serial.println("Stopping A2DP");          
      a2dp_sink.pause();
      bt_active = false;      
      // setup player
      player.setVolume(0.4);
      player.begin();
      Serial.println("Player started");     
  } else {
      Serial.println("Stopping player..");     
      player.end();
      // start sink
      a2dp_sink.play();
      // update sample rate
      auto cfg = i2s.defaultConfig();
      cfg.sample_rate = a2dp_sink.sample_rate();
      i2s.setAudioInfo(cfg);
      Serial.println("A2DP started");
      bt_active = true;   
  }
}

void setup() {
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial, AudioLogger::Info);

  // setup output
  auto cfg = i2s.defaultConfig(TX_MODE);
  cfg.pin_bck = 26;
  cfg.pin_ws = 25;
  cfg.pin_data = 22;
  i2s.begin(cfg);

  // setup player
  source.setFileFilter("*Fireball*"); //my effect

  a2dp_sink.start("MyMusic");
}

void loop() {
  if (player) {
    player.copy();
  } else {
    if (!bt_active){
      changeMode();
    }
  }
  delay(5000);
  changeMode();
}
```

**Comment by pschatzmann** [comment, marked as answer]
Make sure that you deactivate the I2S interface from the BluetoothA2DPSink and manage the audio output yourself via the I2SStream defined in your sketch!
You are currently using 2 different APIs which use the same I2S resource but one does not know about the other. This is just asking for conflicts....
